nomenclature/cli.py

```diff
@@ -93,3 +96,32 @@ def cli_valid_project(
     """
     assert_valid_yaml(path)
     assert_valid_structure(path, definitions, mappings, required_data, dimensions)
+
+
+@cli.command("run-region-processing")
+@click.argument("input_data_file", type=click.Path(exists=True, path_type=Path))
+@click.option("-p", "path", type=click.Path(exists=True, path_type=Path), default=".")
+@click.option("-d", "--definitions", type=str, default="definitions")
+@click.option("-m", "--mappings", type=str, default="mappings")
+@click.option("--output", type=click.Path(path_type=Path), default="results.xlsx")
+@click.option("--export-differences", type=click.Path(path_type=Path), default=None)
+def run_region_processing(
+    input_data_file: Path,
+    path: Path,
+    definitions: str,
+    mappings: str,
+    output: Optional[Path],
+    export_differences: Optional[Path],
+):
+    data_structure_definition = DataStructureDefinition(path / definitions)
+    region_processor = RegionProcessor.from_directory(
+        path / mappings, data_structure_definition
+    )
+    region_processor.return_aggregation_difference = True
+
+    result, difference = region_processor.apply(IamDataFrame(input_data_file))
+
+    if output:
+        result.to_excel(output)
+    if export_differences:
+        difference.reset_index().to_excel(export_differences, index=False)
```

Comment:
You can use the pyam-write-sheet utility to get nicer output formatting.

Reply:
Does that work for pandas dataframes as well?

Reply:
Yes, see https://github.com/IAMconsortium/pyam/blob/1f33e21468d67d365984d0dd7ec44de30c6c60be/pyam/utils.py#L102

But I just realize that it requires a `pd.ExcelWriter`, not just a string to the path, so maybe not worth the effort...

Reply:
Tried getting it to work with `pyam.write_sheet` but for some reason the output file is empty. I'll leave it as is.